### ISSUE TITLE
Firefox SingleDatePicker withPortal not usable

### ISSUE THREAD

**Issue by codepunkt** [airbnb/react-dates#340]
On Firefox the `SingleDatePicker` with the `withPortal` option triggers `onFocusChange` twice, once with `{ focused: true }` and right afterwards `{ focused: false }`. This makes the portal open and close in a matter of half a second, rendering the datepicker unusable.

```javascript
import { SingleDatePicker } from 'react-dates';

export default ({ date, focused, onFocusChange, onDateChange }) => (
    <SingleDatePicker
      id="date_input"
      withPortal
      date={date}
      focused={focused}
      onDateChange={onDateChange}
      onFocusChange={onFocusChange}
      isOutsideRange={(day) => day.isAfter(moment())}
      displayFormat="DD.MM.YYYY"
    />
);
```

Reproduced on Firefox Developer Edition 53.0a2 and Firefox 51.0.1.

**Comment by SiqiTian-minted** [comment]
I'm having a similar issue, but with Chrome when force device to mobile. Besides the double firing of `onFocus`, I also see this error on init:

```
Uncaught (in promise) TypeError: Cannot read property 'getBoundingClientRect' of undefined
    at SingleDatePicker.responsivizePickerPosition (SingleDatePicker.js:443)
    at SingleDatePicker.componentDidMount (SingleDatePicker.js:258)
    at ReactCompositeComponent.js:264
    at measureLifeCyclePerf (ReactCompositeComponent.js:74)
    at ReactCompositeComponent.js:263
    at CallbackQueue.notifyAll (CallbackQueue.js:67)
    at ReactReconcileTransaction.close (ReactReconcileTransaction.js:81)
    at ReactReconcileTransaction.closeAll (Transaction.js:204)
    at ReactReconcileTransaction.perform (Transaction.js:151)
    at ReactUpdatesFlushTransaction.perform (Transaction.js:138)
```

But none of the problem happens on desktop view.

**Comment by jonathanong** [comment]
weird issue... i found that the overlay is shown on firefox if i hold down the left click button or just right click, but then i get the context menu

**Comment by rafacianci** [comment]
I having the same problem.. I set `withPortal: false` and `withFullScreenPortal: true` then I changed the closed button to have it working.. 😓

**Comment by sam-cook** [comment]
Same here, a fix would be awesome

**Comment by lukefrake** [comment]
Yeah, the same issue in Firefox here 😢

**Comment by phamcharles** [comment]
Does anyone have an idea how to fix this? Can we figure out together how to fix it and make a PR?

**Comment by majapw** [comment]
Let me look into it today!

**Comment by phamcharles** [comment]
@majapw This also happens for the daterangepicker on firefox.

**Comment by majapw** [comment]
This should be fixed in v9.0.1!

**Comment by phamcharles** [comment]
@majapw thank you for your hard work and the quick turnaround. I really appreciate it!
